evm_arithmetization/src/generation/state.rs

```diff
@@ -376,15 +376,13 @@ pub struct GenerationState<F: RichField> {
     /// j in [i, i+32] it holds that code[j] < 0x7f - j + i.
     pub(crate) jumpdest_table: Option<HashMap<usize, Vec<usize>>>,
 
-    /// Each entry contains the pair (key, ptr) where key is the (hashed) key
-    /// of an account in the accounts linked list, and ptr is the respective
-    /// node address in memory.
-    pub(crate) accounts_pointers: BTreeMap<U256, usize>,
+    // Provides quick access to pointers that reference the memory location
+    // of an accounts or storage access list node containing a specific key.
+    pub(crate) access_lists_ptrs: LinkedListPtrs,
 
-    /// Each entry contains the pair ((account_key, slot_key), ptr) where
-    /// account_key is the (hashed) key of an account, slot_key is the slot
-    /// key, and ptr is the respective node address in memory.
-    pub(crate) storage_pointers: BTreeMap<(U256, U256), usize>,
+    // Provides quick access to pointers that reference the memory location
+    // of a accounts or storage linked list node containing a specific key.
```

Comment:
/// Provides quick access to pointers that reference the memory location of either:
    /// - an account
    /// - a storage linked-list node containing a specific key.